File: random/random/random.cpp
#include<iostream>
#include<time.h>

using namespace std;

int main() {
	srand((unsigned int)time(0));

	cout << "random : " << (rand() % 100 + 1) << endl;

	double percent;
	percent = rand() % 10000 / 100.f;
	cout << "random : " << percent << endl;

	if(percent <= 80.f)
		cout << "in 80%" << endl;
	if (percent <= 60.f)
		cout << "in 60%" << endl;
	if (percent <= 40.f)
		cout << "in 40%" << endl;
	if (percent <= 20.f)
		cout << "in 20%" << endl;
	if (percent <= 1.f)
		cout << "in 1%" << endl;
	return 0;
}

File: go/random/random.go
package main

import (
	crypto "crypto/rand"
	"fmt"
	"math/rand"
	"time"
)

func main() {
	fmt.Println(rand.Intn(100))

	fmt.Println(rand.Float64())

	fmt.Println((rand.Float64() * 5) + 5)

	timeSource := rand.NewSource(time.Now().UnixNano())
	random := rand.New(timeSource)
	fmt.Println(random.Intn(100))

	cryptoValue := make([]byte, 1)
	crypto.Read(cryptoValue)
	fmt.Println(cryptoValue[0])

	firstSource := rand.NewSource(2018)
	firstRandom := rand.New(firstSource)
	fmt.Println(firstRandom.Intn(100))

	secondSource := rand.NewSource(2018)
	secondRandom := rand.New(secondSource)
	fmt.Println(secondRandom.Intn(100))
}
